contracts/cw1-subkeys/src/contract.rs

```diff
@@ -31,8 +31,11 @@ pub fn init<S: Storage, A: Api, Q: Querier>(
         contract: CONTRACT_NAME.to_string(),
         version: CONTRACT_VERSION.to_string(),
     };
-    set_contract_version(&mut deps.storage, &version)?;
-    whitelist_init(deps, env, msg)
+    let result = whitelist_init(deps, env, msg);
+    if result.is_ok() {
```

Comment:
Nice one, only set on success. 

I guess

```rust
let result = whitelist_init(deps, env, msg)?;
set_contract_version(&mut deps.storage, &version)?;
Ok(result);
```

would be the same, but this works as well